frame/society/src/lib.rs

```diff
@@ -912,33 +1035,50 @@ pub mod pallet {
 		/// The dispatch origin for this call must be _Signed_ and a member with
 		/// payouts remaining.
 		///
-		/// ## Complexity
-		/// - O(M + logM + P + X)
-		///   - M (len of members)
-		///   - P (number of payouts for a particular member)
-		///   - X (currency transfer call)
+		/// Key: M (len of members), P (number of payouts for a particular member)
+		/// Total Complexity: O(M + logM + P + X)
 		#[pallet::call_index(6)]
-		#[pallet::weight(T::BlockWeights::get().max_block / 10)]
+		#[pallet::weight(T::WeightInfo::payout())]
 		pub fn payout(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
+			ensure!(
+				Members::<T, I>::get(&who).ok_or(Error::<T, I>::NotMember)?.rank == 0,
+				Error::<T, I>::NoPayout
+			);
+			let mut record = Payouts::<T, I>::get(&who);
 
-			let members = <Members<T, I>>::get();
-			ensure!(Self::is_member(&members, &who), Error::<T, I>::NotMember);
-
-			let mut payouts = <Payouts<T, I>>::get(&who);
-			if let Some((when, amount)) = payouts.first() {
+			if let Some((when, amount)) = record.payouts.first() {
 				if when <= &<frame_system::Pallet<T>>::block_number() {
+					record.paid = record.paid.checked_add(amount).ok_or(Overflow)?;
 					T::Currency::transfer(&Self::payouts(), &who, *amount, AllowDeath)?;
-					payouts.remove(0);
-					if payouts.is_empty() {
-						<Payouts<T, I>>::remove(&who);
-					} else {
-						<Payouts<T, I>>::insert(&who, payouts);
-					}
+					record.payouts.remove(0);
+					Payouts::<T, I>::insert(&who, record);
 					return Ok(())
 				}
 			}
-			Err(Error::<T, I>::NoPayout.into())
+			Err(Error::<T, I>::NoPayout)?
+		}
+
+		/// Repay the payment previously given to the member with the signed origin, remove any
+		/// pending payments, and elevate them from rank 0 to rank 1.
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::waive_repay())]
+		pub fn waive_repay(origin: OriginFor<T>, amount: BalanceOf<T, I>) -> DispatchResult {
```

Comment:
Could we get time complexity for this one as well since we have it on the other calls?

Reply:
Or the other way, I find most of these complexity/weight comments pretty useless. Even if they should exist, it should be in the benchmarking code, not in the call documentation (where end users will see it through metadata).

Reply:
Indeed - they're actually mostly old and incorrect. As can be seen from the weight calls, they're all O(1) now. The only caveat to that is that `BoundedVec`s in storage items must generally be full to ensure the maximum possible weight is determined. But I don't think that's especially useful to have in the function docs since it's a) fairly obvious; and b) as closely related to the storage item itself as it is the function.